### PR TITLE
Extend Masq support and add unit testing

### PR DESCRIPTION
**Pull request by sbezverk**

### CODE REVIEW

**Comment by sbezverk** [comment]
@stapelberg Please take a look at this PR when you have a moment, thanks a lot..

File: expr/expr.go [hunk 1]
```diff
@@ -135,17 +134,78 @@ func (e *Meta) unmarshal(data []byte) error {
...
+	RegProtoMin uint32
+	RegProtoMax uint32
+}
 
 func (e *Masq) marshal() ([]byte, error) {
+	msgData := []byte{}
+	if !e.ToPorts {
+		flags := uint32(0)
+		if e.Random {
+			flags |= 0x4
```
Comment (stapelberg) [inline expr/expr.go]: Can you use the named constants instead of these values please? :)

File: nftables_test.go [hunk 2]
```diff
@@ -2386,3 +2386,158 @@ func TestSet4(t *testing.T) {
...
+
+func TestMasq(t *testing.T) {
+	tests := []struct {
+		name      string
+		chain     *nftables.Chain
+		want      [][]byte
+		masqExprs []expr.Any
+	}{
+		{
+			name: "Masquerada",
```
Comment (stapelberg) [inline nftables_test.go]: Here and below, correct “Masquerada” to “Masquerade”

Comment (sbezverk) [reply nftables_test.go]: done

File: expr/expr.go (hunk 1 above)
Comment (sbezverk) [reply expr/expr.go]: done

File: expr/expr.go [hunk 3]
```diff
@@ -143,18 +143,27 @@ type Masq struct {
...
 }
 
+const (
+	// MasqRandom defines flag for a random masquerade
+	MasqRandom = 0x4
+	// MasqFullyRandom defines flag for a fully random masquerade
+	MasqFullyRandom = 0x10
+	// MasqPersistent defines flag for a persistent masquerade
+	MasqPersistent = 0x8
+)
```
Comment (stapelberg) [inline expr/expr.go]: Ah, I didn’t know these weren’t available in x/sys/unix yet.

Can you make the names and definition match the `NF_NAT_RANGE_*` constants in `linux/netfilter/nf_nat.h` please?

Comment (sbezverk) [reply expr/expr.go]: done, but names in nf_nat.f looks confusing as no visible relation to masq.

Comment (stapelberg) [reply expr/expr.go]: Yeah, masq re-uses these flags under the hood apparently. Can you add a comment to switch to constants in the unix package once they become available?

**Comment by sbezverk** [comment]
@stapelberg done

**Pull request merged**